index.js

```diff
@@ -311,12 +312,12 @@ module.exports = function AsyncAppendOnlyLog(filename, opts) {
     const blockStart = blockIndex * blockSize
     const blockBuf = blocksWithDeletables.get(blockIndex)
     blocksWithDeletables.delete(blockIndex)
-    blocksWithDeletables.set(-1, null) // indicate that flush is active
+    flushingDelete = true
```

Comment:
This `set(-1, null)` trick seemed smart at the time, but this `null` blockBuf may be causing #87, and booleans are so cheap, we shouldn't be afraid to have one here.

Reply:
right, seems problematic that we can't count on size with that trick.